test/fixtures/iotFleetProvisioning/registerDevice.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+// NOTE: `aws-iot-device-sdk-v2` is bundled into the deployment package
+// eslint-disable-next-line import/no-unresolved
+const { mqtt, io, iot, iotidentity } = require('aws-iot-device-sdk-v2');
+
+const subscriptionCallback = (error, response, channel, resolve, reject) => {
+  if (response) {
+    console.log(`${channel} successful message: ${JSON.stringify(response)}`);
+  }
+
+  if (error || !response) {
+    console.log(`Error occurred on channel ${channel}.`);
+    reject(error);
+  } else {
+    resolve(response);
+  }
+};
+
+const executeKeys = async identity => {
+  return new Promise((resolve, reject) => {
+    console.log('Subscribing to CreateKeysAndCertificate Accepted and Rejected topics..');
+
+    const keysSubRequest = {};
+
+    return Promise.all([
+      identity.subscribeToCreateKeysAndCertificateAccepted(
+        keysSubRequest,
+        mqtt.QoS.AtLeastOnce,
+        (error, response) =>
+          subscriptionCallback(
+            error,
+            response,
+            'subscribeToCreateKeysAndCertificateAccepted',
+            resolve,
+            reject
+          )
+      ),
+      identity.subscribeToCreateKeysAndCertificateRejected(
+        keysSubRequest,
+        mqtt.QoS.AtLeastOnce,
+        (error, response) =>
+          subscriptionCallback(
+            error,
+            response,
+            'subscribeToCreateKeysAndCertificateRejected',
+            resolve,
+            reject
+          )
+      ),
+    ])
+      .then(() => {
```

Comment:
Let's stick to async/await (unless there's a real reason to not do that, like constructing two parallel runs in same function, but it doesn't seem to be the case here)